scripts/batch-score.coffee

```diff
@@ -0,0 +1,107 @@
+# Description:
+#   Script for batch wise score.
+#
+# Configuration:
+#   INFO_SPREADSHEET_URL
+#
+# Commands:
+#   hubot score bxx
+# 
+# Author:
+#   aseem09
+
+module.exports = (robot) ->
+
+  # using this, stores the data in brain's "scorefield" key
+  scorefield = () ->
+    Field = robot.brain.get("scorefield") or {}
+    robot.brain.set("scorefield",Field)
+    Field
+  
+  # response for score status of any <batch>
+  stringLength = (str) ->
+    string=String(str)
+    temp=string.split('')
+    return temp.length
+  
+  pad = (array) ->
+    padding=0
+    max_length=0
+    for name in array
+      max_length= if max_length > stringLength(name) then max_length else stringLength(name)
+    padding = max_length
+    for i in [0..array.length-1]
+      max= padding- stringLength(array[i])
+      if i == 0
+        max=max+3
+      name=array[i]
+      if max>0
+        for j in [0..max-1]
+          name += " "
+      array[i]=name
+    return array
+  
+  robot.respond /score b(\d\d)/i, (msg) ->    
+
+    result=[]
+    temp4= []
+    temp4.push ["Score"]
+    temp5= []
+    temp5.push ["```Name"]
+    sorted=[]
+    response=[]
+
+    # data-store object
+    ScoreField = scorefield()
+
+    #obtaining the current date to calculate relative_year
+    today = new Date
+    mm=today.getMonth()+1
+    yyyy=today.getFullYear()
+    temp1= `yyyy%100`
+    if `mm < 7`
+      `relative_year= temp1 + 4`
+    else
+      `relative_year= temp1 + 5`  
+
+    # <batch> whose score is to be shown
+    batch = msg.match[1]
+    year= relative_year - batch
+
+    robot.http(process.env.INFO_SPREADSHEET_URL)
+      .query({
+        output: "csv"
+      })
+      .get() (err, res, body) ->
+        result = parse body
+        for user in result
+          temp2=user[4]
+          user_year=temp2.split('')
+          temp3 = parseInt(user_year[0], 10 );
+          if `temp3 == year`
+            if user[10]
+              temp4.push [user[10]]
+              temp5.push [user[0]]
+        temp6=pad temp5
+        sorted.push [temp6[0],temp4[0]]
+        for i in [1..temp4.length-1]
+          ScoreField[temp4[i]] = ScoreField[temp4[i]] or 0
+          currentscore = ScoreField[temp4[i]]
+          temp4[i]= currentscore
+          sorted.push [temp6[i],temp4[i]]
+        if sorted.length
+          sorted.sort (a, b) ->
+            b[1] - a[1]
+        sorted = sorted.map (val) -> "#{val[0]} : #{val[1]}"
+        response += sorted.join '\n'
+        response += "```"
+        msg.send response
+
+parse = (json) ->
```

Comment:
It is a common practice to declare your functions before you use. Although this will not raise error(s) but it's a nice practice to follow since people reading your code may stumble if they come across a function which is not in `stdlib` or not yet defined.